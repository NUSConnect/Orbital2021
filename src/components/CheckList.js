import React from 'react'
import {
  StyleSheet,
  Text,
  View,
  Switch,
  Platform
} from 'react-native'
import CheckBox from '@react-native-community/checkbox'

export default class CheckList extends React.Component {
  constructor (props) {
    super()
    this.state = { value: props.item, checked: false }
  }

  componentDidMount () {
    this.setState({ checked: this.props.checked })
    console.log(Platform.OS)
  }

  UNSAFE_componentWillReceiveProps (nextProps) {
    if (nextProps.item !== this.props.item) { this.setState({ ...this.state, value: nextProps.item, checked: false }) }
  }

  handleCheckBoxClick = e => {
    this.props.selectItem(this.state.checked, this.props.item)
    this.setState({
      checked: !this.state.checked
    })
  };

  render () {
    const text = this.state.checked
      ? (
        <Text>{this.state.value}</Text>
        )
      : (
          this.state.value
        )
    return (
      <View id='list' style={styles.widgetUl}>
        <View
          className='main'
          style={styles.main}
        >
          {Platform.OS === 'android'
            ? (
              <CheckBox
                style={{ marginLeft: 5 }}
                onValueChange={this.handleCheckBoxClick}
                value={this.state.checked}
                testID='checkbox'
              />
              )
            : (
              <Switch
                onValueChange={this.handleCheckBoxClick}
                value={this.state.checked}
                testID='switch'
              />
              )}

          <Text style={styles.textBd}>{text}</Text>
        </View>
      </View>
    )
  }

  componentWillUnmount () {}
}

const styles = StyleSheet.create({
  widgetUlX: {
    padding: 0,
    margin: 0,
    display: 'flex',
    justifyContent: 'space-around',
    flexDirection: 'row',
    flexWrap: 'wrap',
    alignItems: 'center'
  },
  widgetUl: {
    paddingVertical: 8,
    width: '50%',
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center'
  },
  textBd: {
    fontSize: 20
  },
  main: {
    flex: 11,
    flexBasis: 250,
    flexDirection: 'row'
  }
})
