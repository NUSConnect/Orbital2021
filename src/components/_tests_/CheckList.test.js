import * as React from 'react'
import CheckList from '../CheckList'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer.create(<CheckList />)
  expect(tree).toMatchSnapshot()
})
