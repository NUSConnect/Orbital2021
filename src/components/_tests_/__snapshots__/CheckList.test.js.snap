// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  id="list"
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "paddingVertical": 8,
      "width": "50%",
    }
  }
>
  <View
    className="main"
    style={
      Object {
        "flex": 11,
        "flexBasis": 250,
        "flexDirection": "row",
      }
    }
  >
    <RCTSwitch
      accessibilityRole="switch"
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": 31,
          "width": 51,
        }
      }
      value={false}
    />
    <Text
      style={
        Object {
          "fontSize": 20,
        }
      }
    />
  </View>
</View>
`;
